### DISCUSSION TITLE
[Atari] SpartaDOS X RealTime Clock Access

### DISCUSSION THREAD

**Discussion by GSoftwareDevelopment** [cc65/cc65#2118] [Q&A] [answered]
Hi.

I have a problem with reading the current time and date. Generally I'm weak in C and I'm kind of doing everything at random, however.... something is wrong.

The theory goes that `puts(ctime(CLOCK_REALTIME));` should display the current date and time on the screen, but I get an undefined garbage disposal.

I point out that I have SDX configured correctly and that the RTC is in the system.

I would greatly appreciate some guidance on how to read the current date and time.

As an additional note, let's say that I want to use the date and time reading under Contiki.

**Comment by mrdudz** [comment, marked as answer]
That is not how it works, see https://www.educative.io/answers/what-is-ctime-in-c

**Comment by GSoftwareDevelopment** [comment]
Beautifully. It worked from the kick (so to speak)
The downside is that quite time-consuming operation and not suitable for inserting in the log :(

Anyway, thank you very much - somehow I missed, searching for the description of `ctime` :)

**Reply by polluks** [reply]
How about accessing drive Z:?

**Reply by GSoftwareDevelopment** [reply]
Maybe the issue of the time-consuming operation, was the fact that the tests took place on an emulator (?) I do not know, but on real hardware it turned out to be completely unconflicted.

Access via Z drive:... I find it more comfortable to operate at the SDX system level, given that not everyone needs to have the Z driver installed: :) In any case, this is a good option.
